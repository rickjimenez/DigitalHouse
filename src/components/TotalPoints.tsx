import React, {FC} from 'react';
import {Text, View, StyleSheet, ActivityIndicator} from 'react-native';
import {GetProductsResponse} from '../types/api';
import {useGlobalStyles} from '../styles/global';

interface Props {
  data: GetProductsResponse[] | undefined;
  isLoading?: boolean;
}

/**
 * TotalPoints Component
 *
 * @returns {React.ReactElement} TotalPoints.
 */
const TotalPoints: FC<Props> = ({
  data = [],
  isLoading = false,
}): JSX.Element => {
  const styles = useStyles();
  const globalStyles = useGlobalStyles();

  const total = data.reduce((acc, cur) => {
    if (cur.is_redemption) {
      acc -= cur.points;
    } else {
      acc += cur.points;
    }
    return acc;
  }, 0);

  return (
    <View>
      <Text style={globalStyles.title}>TUS PUNTOS</Text>
      <View style={styles.container}>
        {isLoading ? (
          <ActivityIndicator animating={true} color="white" />
        ) : (
          <>
            <Text style={[styles.text, styles.subtitle]}>Diciembre</Text>
            <View style={styles.points}>
              <Text style={[styles.text, styles.pointsTxt]}>{total} pts</Text>
            </View>
          </>
        )}
      </View>
    </View>
  );
};

const useStyles = () =>
  StyleSheet.create({
    container: {
      width: 286,
      height: 143,
      backgroundColor: '#334FFA',
      borderRadius: 20,
      shadowColor: '#000',
      shadowOffset: {
        width: 0,
        height: 2,
      },
      shadowOpacity: 0.25,
      shadowRadius: 3.84,

      elevation: 5,
      padding: 20,
      alignSelf: 'center',
      justifyContent: 'center',
      marginBottom: 20,
    },
    points: {
      alignItems: 'center',
      justifyContent: 'center',
      flex: 1,
    },
    text: {
      color: 'white',
      fontWeight: 'bold',
    },
    subtitle: {
      fontSize: 16,
    },
    pointsTxt: {
      fontSize: 32,
    },
  });

export default TotalPoints;
