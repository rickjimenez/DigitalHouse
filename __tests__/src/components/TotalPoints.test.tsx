import 'react-native';
import {render, screen} from '@testing-library/react-native';
import React from 'react';
import TotalPoints from '../../../src/components/TotalPoints';

const data = [
  {
    id: '1',
    product: 'product',
    is_redemption: false,
    createdAt: '2022-12-09T06:34:25.607Z',
    image: 'image',
    points: 100,
  },
  {
    id: '1',
    product: 'product',
    is_redemption: true,
    createdAt: '2022-12-09T06:34:25.607Z',
    image: 'image',
    points: 30,
  },
];

describe('test TotalPoints Comp', () => {
  test('renders correctly', () => {
    render(<TotalPoints data={data} />);

    const comp = screen.toJSON();
    expect(comp).toMatchSnapshot();
  });

  test('test total points', () => {
    render(<TotalPoints data={data} />);
    const result = screen.getByText('70 pts');
    expect(result).toBeTruthy();
  });
});
