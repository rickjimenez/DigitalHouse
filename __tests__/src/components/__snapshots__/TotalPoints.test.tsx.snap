// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test TotalPoints Comp renders correctly 1`] = `
<View>
  <Text
    style={
      {
        "color": "#9B9898",
        "fontSize": 14,
        "fontWeight": "800",
        "marginBottom": 10,
      }
    }
  >
    TUS PUNTOS
  </Text>
  <View
    style={
      {
        "alignSelf": "center",
        "backgroundColor": "#334FFA",
        "borderRadius": 20,
        "elevation": 5,
        "height": 143,
        "justifyContent": "center",
        "marginBottom": 20,
        "padding": 20,
        "shadowColor": "#000",
        "shadowOffset": {
          "height": 2,
          "width": 0,
        },
        "shadowOpacity": 0.25,
        "shadowRadius": 3.84,
        "width": 286,
      }
    }
  >
    <Text
      style={
        [
          {
            "color": "white",
            "fontWeight": "bold",
          },
          {
            "fontSize": 16,
          },
        ]
      }
    >
      Diciembre
    </Text>
    <View
      style={
        {
          "alignItems": "center",
          "flex": 1,
          "justifyContent": "center",
        }
      }
    >
      <Text
        style={
          [
            {
              "color": "white",
              "fontWeight": "bold",
            },
            {
              "fontSize": 32,
            },
          ]
        }
      >
        70
         pts
      </Text>
    </View>
  </View>
</View>
`;
